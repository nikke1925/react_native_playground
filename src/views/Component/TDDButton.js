// @flow
import React, { Component } from 'react';
import { View, StyleSheet, Text, TouchableOpacity } from 'react-native';

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#C5EFF7',
    flex: 1,
  },
});

export default class TDDButton extends Component {
  render() {
    return (
      <View style={styles.container}>
        <TouchableOpacity>
          <Text>TDD-Button</Text>
        </TouchableOpacity>
      </View>
    );
  }
}
