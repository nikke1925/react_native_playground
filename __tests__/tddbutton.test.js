// @noflow
import React from 'react';
import renderer from 'react-test-renderer';
import TDDButton from '../src/views/Component/TDDButton';

it('renders correctly', () => {
  const tree = renderer.create(<TDDButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
