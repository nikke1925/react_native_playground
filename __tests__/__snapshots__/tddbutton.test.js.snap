// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#C5EFF7",
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      TDD-Button
    </Text>
  </View>
</View>
`;
